### ISSUE TITLE
opMul is deprecated and won't work

### ISSUE THREAD

**Issue by trikko** [aferust/opencvd#8]
opMul() is used in cvcore, but it is deprecated and it won't work anymore.

**Comment by aferust** [comment]
I ve just committed some housekeeping changes 2e36aa5204a4d6010cc4311d7e982b17fdbb5dd3. Please confirm if those changes works for your condition.

Thank you for the contribution!

**Comment by trikko** [comment]
Yes!

**Comment by aferust** [comment]
There is a discussion in the forum that c style struct initialization will be deprecated? I used them in opencvd a lot. Hope it will not happen in the near future. So that means a lot of work! I don't know what I was thinking by doing it.

**Comment by trikko** [comment]
Probably it is not going to happen. It is related to named params and I think it won't go anywhere.
Anyway regex could help in that case :)

**Comment by aferust** [comment]
Please pull the repo again 'cause I ve fixed some typos in opBinary.

**Comment by Geod24** [comment]
@aferust : I doubt struct literal are going anywhere anytime soon. The alternative is not just filling the gap they would leave.

I just tried this project and noticed that the latest release (`v0.0.5` ) does not compile:
```
../../../.dub/packages/opencvd-0.0.5/opencvd/source/opencvd/cvcore.d(791,9): Error: function opencvd.cvcore.Mat.opBinary!"*".opBinary has no return statement, but is expected to return a value of type Mat
source/app.d(84,13): Error: template instance opencvd.cvcore.Mat.opBinary!"*" error instantiating
```

But I suppose master is fixed.
Also, side question: Did you consider binding the C++ API ? Did you just pick the C API because it would be more stable / simple ?

**Comment by aferust** [comment]
I don't know which example raised that error. Yes ~master should compile. I am not sure we have enough changes to deploy a new release for dub.

When I started learning D, I started to make this binding in the same time. There should be some rookie code that sometimes I detect and rewrite over time as my knowledge increases. So creating a binding from C was pretty much easier for me. A C++ based binding would be a new project that I don't have motivation and time for it nowadays. I think the difficult part should be wrapping cv::Mat. I would contribute to a new project happily though.
